### PR TITLE
Adding loader for BAF dataset

### PR DESCRIPTION
**Pull request by guillemcortes**
### Adding loader for BAF

 Please use the following title: "Adding loader for MyDATASET". If your pull request is work in progress, change your title to "[WIP] Adding loader for MyDATASET" to avoid reviews while the loader is not ready.

### Description

Please include the following information at the top level docstring for the dataset's module mydataset.py:

- [x] Describe annotations included in the dataset
- [x] Indicate the size of the datasets (e.g. number files and duration, hours)
- [x] Mention the origin of the dataset (e.g. creator, institution)
- [x] Describe the type of music included in the dataset
- [x] Indicate any relevant papers related to the dataset
- [x] Include a description about how the data can be accessed and the license it uses (if applicable)

#### Dataset loaders checklist:

- [x] Create a script in `scripts/`, e.g. `make_my_dataset_index.py`, which generates an index file.
- [x] Run the script on the canonical version of the dataset and save the index in `mirdata/indexes/` e.g. `my_dataset_index.json`.
- [x] Create a module in mirdata, e.g. `mirdata/my_dataset.py`
- [x] Create tests for your loader in `tests/datasets/`, e.g. `test_my_dataset.py`
- [x] Add your module to `docs/source/mirdata.rst` and `docs/source/table.rst`
- [x] Run `tests/test_full_dataset.py` on your dataset.

If your dataset **is not fully downloadable** there are two extra steps you should follow:
- [x] Contacting the mirdata organizers by opening an issue or PR so we can discuss how to proceed with the closed dataset.
- [x] Show that the version used to create the checksum is the "canonical" one, either by getting the version from the dataset creator, or by verifying equivalence with several other copies of the dataset.
- [x] Make sure someone has run `pytest -s tests/test_full_dataset.py --local --dataset my_dataset` once on your dataset locally and confirmed it passes

#### Please-do-not-edit flag
To reduce friction, we will make commits on top of contributor's pull requests by default unless they use the `please-do-not-edit` flag. If you don't want this to happen don't forget to add the flag when you start your pull request.

### CODE REVIEW

File: tests/datasets/test_baf.py [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+    run_track_tests(track, expected_attributes, expected_property_types)
+
+
+def test_to_jams():
+    default_trackid = TRACK_ID
+    dataset = baf.Dataset(TEST_DATA_HOME)
+    track = dataset.track(default_trackid)
+    jam = track.to_jams()
+
+    # Validate cante100 jam schema
```
Comment (genisplaja) [inline tests/datasets/test_baf.py:46]: Remove or update (jeje cante100 ftw)

File: mirdata/datasets/baf.py [hunk 2]
```diff
@@ -0,0 +1,436 @@
...
+
+@core.docstring_inherit(annotations.EventData)
+class EventDataExtended(annotations.EventData):
+    """EventDataExtended class
+
+    Attributes:
+        tags (list): list of tag labels (as strings)
+        tag_unit (str): tag units, one of TAG_UNITS
+
+    """
```
Comment (genisplaja) [inline mirdata/datasets/baf.py:210]: Looks good, and I personally like the idea of this class here. However, I think it'd be good to briefly present in the docstring what is the difference between the original `EventData` and the extended version here! Just in a single sentence would be enough :) 
That class is a nice example for the issue in #558.

File: mirdata/datasets/baf.py [hunk 3]
```diff
@@ -0,0 +1,436 @@
...
+        try:
+            metadata_df = pd.read_csv(metadata_path)
+        except FileNotFoundError as not_found:
+            raise FileNotFoundError(
+                FILENOTFOUND_MSG.safe_substitute(fname=not_found.filename)
+            )
+        try:
+            xannotations_df = pd.read_csv(xannotations_path)
+        except FileNotFoundError as not_found:
+            raise FileNotFoundError(
```
Comment (genisplaja) [inline mirdata/datasets/baf.py:393]: This is not covered by the tests. I guess you could do something in the tests like:
```
with pytest.raises(FileNotFoundError):
    feat = baf._metadata(/a/fake/path)
```

**Review by genisplaja — COMMENTED** [review]
Hey @guillemcortes, I am just leaving a few tiny comments but the loader looks really nice. Good coverage, docs rendering correctly, and tests passing :) I'll invite @nkundiushuti, @magdalenafuentes, and @harshpalan to take a look, and IMO we can merge!

**Comment by guillemcortes** [comment]
Hi @genisplaja, thanka for your comments, I've done the according modifications. I've also improved the csv2pandas management and atomized the function. I think now it's better. I've also included the corresponding test. Would you mind checking it? Thanks!

**Comment by genisplaja** [comment]
That's nice! Thanks @guillemcortes :) Maybe @harshpalan or @magdalenafuentes could take a look? 🙏🏼🙏🏼🙏🏼

**Review by harshpalan — APPROVED** [review]
Hi, @guillemcortes @genisplaja sorry for the delayed response. This looks good. I'm approving the changes. Thanks.

**Pull request merged**